jb/itch5/compute_book_depth.cpp

```diff
@@ -0,0 +1,195 @@
+#include <jb/itch5/compute_book_depth.hpp>
+#include <jb/assert_throw.hpp>
+
+jb::itch5::compute_book_depth::compute_book_depth(callback_type const& cb)
+    : callback_(cb)
+    , orders_()
+    , books_()
+{}
+
+void jb::itch5::compute_book_depth::handle_message(
+    time_point recv_ts, long msgcnt, std::size_t msgoffset,
+    stock_directory_message const& msg) {
+  // Only log these messages if we want super-verbose output, there
+  // are nearly 8,200 securities in the Nasdaq exchange.
+  JB_LOG(trace) << " " << msgcnt << ":" << msgoffset << " " << msg;
+  // ... create the book and update the map ...
+  books_.emplace(msg.stock, order_book_depth());
+}
+
+void jb::itch5::compute_book_depth::handle_message(
+    time_point recv_ts, long msgcnt, std::size_t msgoffset,
+    add_order_message const& msg) {
+  JB_LOG(trace) << " " << msgcnt << ":" << msgoffset << " " << msg;
+  auto r = handle_add_order(recv_ts, msgcnt, msgoffset, msg);
+  if (std::get<0>(r)) {
```

Comment:
Hmm... I think we agreed to update on all events, not just the ones that changed the depth or the inside, right?  What do you need this if() statement if that is the case?